src/torrent.cpp

```diff
@@ -9175,7 +9175,11 @@ namespace {
 		int seeds = 0;
 		int downloaders = 0;
 
-		if (m_complete != 0xffffff) seeds = m_complete;
+		// If we're currently seeding and using tracker supplied scrape
+		// data, we should remove ourselves from the seed count
+		int const self_seed = is_seed() && !is_paused() ? 1 : 0;
```

Comment:
I think this should be `std::uint32_t` rather than `int`. to be compatible with `m_complete`.

https://github.com/arvidn/libtorrent/pull/6311/checks?check_run_id=3032982089#step:6:202

```
..\src\torrent.cpp(9182): error C2672: 'std::max': no matching overloaded function found
203
..\src\torrent.cpp(9182): error C2780: '_Ty std::max(std::initializer_list<_Elem>)': expects 1 arguments - 2 provided
204
C:\Program Files (x86)\Microsoft Visual Studio\2019\Enterprise\VC\Tools\MSVC\14.29.30037\include\utility(52): note: see declaration of 'std::max'
205
..\src\torrent.cpp(9182): error C2784: '_Ty std::max(std::initializer_list<_Elem>,_Pr)': could not deduce template argument for 'std::initializer_list<_Elem>' from 'int'
206
C:\Program Files (x86)\Microsoft Visual Studio\2019\Enterprise\VC\Tools\MSVC\14.29.30037\include\utility(49): note: see declaration of 'std::max'
207
..\src\torrent.cpp(9182): error C2782: 'const _Ty &std::max(const _Ty &,const _Ty &) noexcept(<expr>)': template parameter '_Ty' is ambiguous
208
C:\Program Files (x86)\Microsoft Visual Studio\2019\Enterprise\VC\Tools\MSVC\14.29.30037\include\utility(41): note: see declaration of 'std::max'
209
..\src\torrent.cpp(9182): note: could be 'uint32_t'
210
..\src\torrent.cpp(9182): note: or       'int'
```

Reply:
oh, now the issue is that `0` is `int`. You might have to say `std::uint32_t(0)`

Reply:
Third time lucky...

Reply:
Or maybe even fourth

Reply:
I went the other way and converted `m_complete` into an `int` instead. The other relevant vars (`downloaders`, `scale`, `seeds`) are also `int`, and it seems easier than trying to get my head around implicit conversion and signedness.

Looks like I have some reading to do.